### ISSUE TITLE
Feedback for “Self-Hosting Hive”

### ISSUE THREAD

**Issue by wethekings303** [kamilkisiela/graphql-hive#3617]
Missing documentation for adding https to hive microservices such as the app and usage services. Is there functionality to support tls or is it only done through a reverse proxy at this time? I would also state that adding trusted self signed CAs for enterprise would be great to add documentation for as well.

For reference I am deploying onto a kubernetes cluster and have nginx ingress but do not want to terminate at the ingress and instead do ssl passthrough to the pod to allow tls termination at the pod so there is no unencrypted communication in our environment.

**Comment by wethekings303** [comment]
Hive cli does not trust self signed certificates as well. The certs are provided by our internal CA and are valid but not publicly trusted like most enterprises that run their own software. Can this be addressed so that our team can properly evaluate the graphql hive product?

**Comment by n1ru4l** [comment]
> Hive cli does not trust self signed certificates as well. The certs are provided by our internal CA and are valid but not publicly trusted like most enterprises that run their own software. Can this be addressed so that our team can properly evaluate the graphql hive product?

This should be achievable via the `NODE_EXTRA_CA_CERTS` environment variable. [See the Node.js documentation](https://nodejs.org/docs/latest-v18.x/api/cli.html#node_extra_ca_certsfile).

```
NODE_EXTRA_CA_CERTS=path hive-cli schema:check ./schema.graphql
```


> Missing documentation for adding https to hive microservices such as the app and usage services. Is there functionality to support tls or is it only done through a reverse proxy at this time? I would also state that adding trusted self signed CAs for enterprise would be great to add documentation for as well.
>
> For reference I am deploying onto a kubernetes cluster and have nginx ingress but do not want to terminate at the ingress and instead do ssl passthrough to the pod to allow tls termination at the pod so there is no unencrypted communication in our environment.

This is not one of our priorities right now.

Quick research showed me some options that allow achieving encrypted traffic between pods using Kubernetes plugins/extensions without having to make it an application/service concern. [See StackOverflow post](https://stackoverflow.com/a/67302048/4202031).

We are also open to accepting contributions for what you are requesting.

**Comment by wethekings303** [comment]
The NODE_EXTRA_CA_CERTS environment variable worked great for both SSO integration and the hive cli. 

We ended up configuring tls on the nginx ingress for now to have a trusted cert at the app layer. I have ran Istio as several past companies but not at the current one. It might be a consideration to enable full encrypted communication for unsupported services.

I might try adding some contributions for tls for the different services I had questions about.

Thanks so much for the reply!
